flask/app/routes.py

```diff
@@ -339,6 +384,8 @@ def bulk_update():
         else:
             files = row.get("linked_files", [])
         dataset.files += [models.DatasetFile(path=path) for path in files if path]
+        for g in user_group:
+            dataset.groups.append(g)
```

Comment:
We need to use the groups specified in the query string if it's provided. This looks like it unconditionally adds the dataset to all of the user's permission groups.

Reply:
ah good point, fixed now!